packages/analytics-core/src/utils/amplitude-promise.ts

```diff
@@ -0,0 +1,5 @@
+export const amplitudePromise =
```

Comment:
Why the rename? I though return wrapper was ok

Reply:
Oh I was under the impression there was confusion on what's returning `.promise` and what's not so I thought of giving it a more obvious name. I can revert though

Reply:
Sorry before I was asking why we couldn't have `Promise<T>` in the `AmplitudeReturn.promise` type def. It looks like you fixed that in this PR 🙌 

I think `AmplitudeReturn` is more flexible and would allow us to return additional data if needed in the future, including synchronously.
```
export interface AmplitudeReturn<T> {
  promise: Promise<T>;
  someSyncronousReturnValue: string;
}
```